ocaml/xapi/helpers.ml

```diff
@@ -1320,6 +1320,20 @@ let rmtree path =
     let msg = Printf.sprintf "failed to remove %s: %s" path exn' in
     failwith msg
 
+let resolve_uri_path ~root ~uri_path =
+  let open Xapi_stdext_std.Xstringext in
+  uri_path
+  |> Filename.concat root
+  |> Uri.pct_decode
+  |> Xapi_stdext_unix.Unixext.resolve_dot_and_dotdot
+  |> (fun x -> match (String.startswith (root^"/") x), (Sys.file_exists x) with
```

Comment:
Please use `Astring.String.is_prefix ~affix:(root ^
"/") x` instead:
```suggestion
  uri_path
  |> Filename.concat root
  |> Uri.pct_decode
  |> Xapi_stdext_unix.Unixext.resolve_dot_and_dotdot
  |> (fun x -> match (Astring.String.is_prefix ~affix:(root ^
"/") x, Sys.file_exists x) with
```

Reply:
I see that this change has now been made.